nat/Cargo.toml

```diff
@@ -5,6 +5,9 @@ edition = "2024"
 publish = false
 license = "Apache-2.0"
 
+[features]
+shuttle = ["concurrency/shuttle", "dep:shuttle"]
```

Comment:
@mvachhar The `shuttle` feature is unused at the moment, right? Why introduce it in a a clean-up commit, without mentioning it (or its future use case) in the commit/PR descriptions?

Reply:
@mvachhar Thinking of this again, I still have doubts. What's the point of this feature? Sure, if you build with `cargo build -F shuttle`, this will pull `concurrency/shuttle` and make the “concurrency mode” be `shuttle`.

_But_ if you use `cargo build -F shuttle`, you _already_ build the `concurrency` crate with feature `shuttle` _anyway_, so what does the local `shuttle` feature bring? Why not keep just one feature in `concurrency`? It's not as if we picked features individually for each crate on the command line, is it?